### PR TITLE
test(autoware_route_handler): add unit test for autoware route handler function

### PR DESCRIPTION
**Pull request by mkquda**
## Description
Improve code coverage of autoware_route_handler:

- Remove unused functions
- add unit test and docstring for function getShoulderLaneletsAtPose
- add test case for getCenterLinePath to improve branch coverage

## How was this PR tested?

run unit tests and generate code coverage report.

before:
Screenshot from 2024-07-31 08-26-49

after:
Screenshot from 2024-07-31 09-09-49

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/autoware_route_handler/include/autoware/route_handler/route_handler.hpp [hunk 1]
```diff
@@ -353,17 +355,6 @@ class RouteHandler
...
-    const lanelet::ConstLanelets & lanelet_section1,
-    const lanelet::ConstLanelets & lanelet_section2) const;
-  bool getPreviousLaneletWithinRouteExceptGoal(
-    const lanelet::ConstLanelet & lanelet, lanelet::ConstLanelet * prev_lanelet) const;
-  bool getNextLaneletWithinRouteExceptStart(
-    const lanelet::ConstLanelet & lanelet, lanelet::ConstLanelet * next_lanelet) const;
-  bool getRightLaneletWithinRoute(
-    const lanelet::ConstLanelet & lanelet, lanelet::ConstLanelet * right_lanelet) const;
-  bool getLeftLaneletWithinRoute(
-    const lanelet::ConstLanelet & lanelet, lanelet::ConstLanelet * left_lanelet) const;
```
Comment (zulfaqar-azmi-t4) [inline planning/autoware_route_handler/include/autoware/route_handler/route_handler.hpp:366]: Is there any reason why these unused code  is removed in the unit test's PR?

File: planning/autoware_route_handler/test/test_route_handler.cpp [hunk 2]
```diff
@@ -289,4 +298,20 @@ TEST_F(TestRouteHandler, testGetLaneChangeTargetLanes)
     ASSERT_EQ(lane_change_lane.value().id(), 9598ul);
   }
 }
+
+TEST_F(TestRouteHandler, testGetShoulderLaneletsAtPose)
```
Comment (zulfaqar-azmi-t4) [inline planning/autoware_route_handler/test/test_route_handler.cpp:302]: Adding comments to inform reader about whether the pose is inside or outside the shoulder lanes would be helpful.

Also, if the pose is outside of the shoulder lanes, what result would it be?

File: planning/autoware_route_handler/include/autoware/route_handler/route_handler.hpp (hunk 1 above)
Comment (mkquda) [reply planning/autoware_route_handler/include/autoware/route_handler/route_handler.hpp:366]: I remember I saw somewhere in the testing guidelines to remove unused member functions ...

**Review by zulfaqar-azmi-t4 — APPROVED** [review]

File: planning/autoware_route_handler/test/test_route_handler.cpp (hunk 2 above)
Comment (zulfaqar-azmi-t4) [reply planning/autoware_route_handler/test/test_route_handler.cpp:302]: Since we're going for line coverage, I'll approve this PR for now!

**Pull request merged**